### PR TITLE
(T): use rust sources to test parser performance

### PR DESCRIPTION
**Pull request by matklad**
This adds the first performance test. The test simply executes the parser on the compiler's source code.

### CODE REVIEW

**Comment by matklad** [comment]
@jajakobyly please take a look

File: src/test/kotlin/org/rust/Performance.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.rust
+
+/**
+ * Marker interface for tests category
+ */
+interface Performance {
+}
```
Comment (jajakobyly) [inline src/test/kotlin/org/rust/Performance.kt]: This braces can be omitted :smile:

File: src/test/kotlin/org/rust/lang/RustTestCaseBase.kt [hunk 2]
```diff
@@ -68,6 +68,16 @@ abstract class RustTestCaseBase : LightPlatformCodeInsightFixtureTestCase(), Rus
...
                 .joinToString("_")
+
+        @JvmStatic
+        fun rustSources(): VirtualFile {
+            val sdkArchiveFile = LocalFileSystem.getInstance()
+                .findFileByPath("${RustTestCase.testResourcesPath}/rustc-src.zip")
+
+            return checkNotNull(sdkArchiveFile?.let { JarFileSystem.getInstance().getJarRootForLocalFile(it) }) {
+                "Rust sources archive not found. Run `./gradlew test` to download the archive."
+            }
```
Comment (jajakobyly) [inline src/test/kotlin/org/rust/lang/RustTestCaseBase.kt]: Pretty much the same thing is also done inside `WithStdlibRustProjectDescriptor#testCargoProject`

**Comment by jajakobyly** [comment]
:+1:

**Comment by jajakobyly** [comment]
It might be nice to run this, let's say, every 10th CI build.

**Comment by matklad** [comment]
It would be run on every CI build. I'm not sure how robust are time measurements though, they are a bit high at the moment. I hope to run `./gradle performanceTest` manually sometimes.

**Comment by matklad** [comment]
It's about time to stop waffling about it :) 

I've removed actual timing assertions because I don't believe they'll well with our current CI infra.

**Comment by matklad** [comment]
Yay, down to single branch again :tada:

**Pull request merged**